src/Sylius/Bundle/CoreBundle/PaymentRequest/Processor/AfterOfflineCaptureProcessor.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\Processor;
+
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\PaymentTransitions;
+use Webmozart\Assert\Assert;
+
+final class AfterOfflineCaptureProcessor implements AfterOfflineCaptureProcessorInterface
+{
+    public function __construct(
+        private StateMachineFactoryInterface|StateMachineInterface $stateMachineFactory,
+    ) {
+        if ($this->stateMachineFactory instanceof StateMachineFactoryInterface) {
+            trigger_deprecation(
+                'sylius/payment-bundle',
+                '1.13',
+                sprintf(
+                    'Passing an instance of "%s" as the first argument is deprecated. It will accept only instances of "%s" in Sylius 2.0.',
+                    StateMachineFactoryInterface::class,
+                    StateMachineInterface::class,
+                ),
+            );
+        }
+    }
+
+    public function process(PaymentRequestInterface $paymentRequest): void
+    {
+        $payment = $paymentRequest->getPayment();
+        Assert::notNull($payment);
+
+        $stateMachine = $this->stateMachineFactory->get($payment, PaymentTransitions::GRAPH);
+        if ($paymentRequest->getResponseData()['paid']) {
+            $stateMachine->apply(PaymentTransitions::TRANSITION_COMPLETE);
+        } else {
+            $stateMachine->apply(PaymentTransitions::TRANSITION_PROCESS);
+        }
```

Comment:
The main problem I see with that is you'd need a matrix of paymentRequest state machines for every factory and paymentRequest type within, which might get messy fast.
Let's say we have 3 factories, each having 4 steps per payment (4 different types of requests), I'd assume you'd need 12 state machines to process the payment's state correctly since each factory can change the state of payment differently per type.
Having to set it manually isn't maybe annoying, but at least the person implementing it can be sure what state will be the result and there is no internal magic that can be potentially buggy.